src/main/java/chess/domain/piece/Piece.java

```diff
@@ -0,0 +1,81 @@
+package chess.domain.piece;
+
+import chess.domain.Position;
+import chess.domain.Score;
+import chess.domain.TeamColor;
+import chess.domain.piece.moving.PieceMoving;
+import chess.exception.ImpossibleMoveException;
+
+import java.util.List;
+
+import static chess.domain.TeamColor.WHITE;
+
+public abstract class Piece {
+
+    private final String name;
+    private final TeamColor teamColor;
+    private final Score score;
+    private final PieceMoving pieceMoving;
+
+    public Piece(String name, TeamColor teamColor, Score score, PieceMoving pieceMoving) {
+        this.name = name;
+        this.teamColor = teamColor;
+        this.score = score;
+        this.pieceMoving = pieceMoving;
+    }
+
+    public void updateMovablePositions(List<Position> existPiecePositions, List<Position> enemiesPositions) {
+        pieceMoving.updateMovablePositions(existPiecePositions, enemiesPositions);
+    }
+
+    public void move(Position targetPosition) throws ImpossibleMoveException {
+        pieceMoving.move(targetPosition);
+    }
+
+    public Position currentPosition() {
+        return pieceMoving.currentPosition();
+    }
+
+    public boolean isSamePosition(Position position) {
+        return pieceMoving.isSamePosition(position);
+    }
+
+    public int row() {
+        return pieceMoving.row();
+    }
+
+    public List<Position> movablePositions() {
+        return pieceMoving.movablePositions();
+    }
+
+    public boolean sameColor(TeamColor teamColor) {
+        return this.teamColor.isSameColor(teamColor);
+    }
+
+    public boolean notSameColor(TeamColor teamColor) {
+        return this.teamColor.isNotSameColor(teamColor);
+    }
+
+    public TeamColor enemyColor() {
+        return teamColor.reverse();
+    }
+
+    public TeamColor teamColor() {
+        return teamColor;
+    }
+
+    public abstract boolean isPawn();
+
+    public abstract boolean isKing();
```

Comment:
```suggestion
    public boolean isPawn() {
        return false;
    }

    public abstract boolean isKing();
```
부모 클래스에서 정의하고 필요한 클래스에서만 재정의 하는 방법도 있습니다 :) 

```java
public class Pawn extends Piece {
  ..
  
  @Override
    public boolean isPawn() {
        return true;
    }
}
```

Reply:
질문의 메서드가 여기네요. 비즈니스에 필요하니 필요한 메서드라고 생각이 들어요. 또한 각 객체에게 묻는 편이 `instanceof` 보다는 훨씬 나은것 같습니다 👍

Reply:
오 이 방법이 훨씬 좋아보이네요 부모 클래스에서 정의하고 필요한 클래스인 King과 Pawn에서만 override하도록 변경했습니다!